paddle/fluid/framework/ir/conv_bn_fuse_pass.cc

```diff
@@ -0,0 +1,331 @@
+                                       batch_norm, bn_mean_out, bn_variance_out,
+                                       bn_saved_mean, bn_saved_variance});
+
+    PADDLE_ENFORCE(subgraph.count(conv_input));
+    IR_NODE_LINK_TO(conv_out, bias_op);
+    IR_NODE_LINK_TO(y_var_node, bias_op);
+    IR_NODE_LINK_TO(bias_op, bn_out);
+
+    found_conv_bn_count++;
+  };
+
+  gpd(graph.get(), handler);
+
+  AddStatis(found_conv_bn_count);
+  return graph;
+}
+
+std::unique_ptr<ir::Graph> ConvEltwiseAddBNFusePass::ApplyImpl(
+    std::unique_ptr<ir::Graph> graph) const {
+  PADDLE_ENFORCE(graph.get());
+  FusePassBase::Init(name_scope_, graph.get());
+
+  auto* scope = param_scope();
+  PADDLE_ENFORCE(scope);
+
+  GraphPatternDetector gpd;
+  auto* conv_input =
+      gpd.mutable_pattern()
+          ->NewNode(patterns::PDNodeName(name_scope_, "conv_input"))
+          ->AsInput()
+          ->assert_is_op_input("conv2d", "Input");
+  patterns::ConvBN conv_bn_pattern(gpd.mutable_pattern(), name_scope_);
+  conv_bn_pattern(conv_input, true /*with_eltwise_add*/);
+
+  int found_conv_bn_count = 0;
+  auto handler = [&](const GraphPatternDetector::subgraph_t& subgraph,
+                     Graph* g) {
+    VLOG(4) << "handle ConvBN fuse";
+    // OPERATORS
+    GET_IR_NODE_FROM_SUBGRAPH(conv, conv, conv_bn_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(bias, bias, conv_bn_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(batch_norm, batch_norm, conv_bn_pattern);
+    // CONV inputs
+    GET_IR_NODE_FROM_SUBGRAPH(conv_weight, conv_weight, conv_bn_pattern);
+    // CONV outputs
+    GET_IR_NODE_FROM_SUBGRAPH(conv_out, conv_out, conv_bn_pattern);
+    // BIAS inputs
+    GET_IR_NODE_FROM_SUBGRAPH(bias_y, bias_y, conv_bn_pattern);
+    // BIAS outputs
+    GET_IR_NODE_FROM_SUBGRAPH(bias_out, bias_out, conv_bn_pattern);
+    // BN inputs
+    GET_IR_NODE_FROM_SUBGRAPH(bn_scale, bn_scale, conv_bn_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(bn_bias, bn_bias, conv_bn_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(bn_mean, bn_mean, conv_bn_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(bn_variance, bn_variance, conv_bn_pattern);
+    // BN outputs
+    GET_IR_NODE_FROM_SUBGRAPH(bn_out, bn_out, conv_bn_pattern);  // Out
+    GET_IR_NODE_FROM_SUBGRAPH(bn_mean_out, bn_mean_out, conv_bn_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(bn_variance_out, bn_variance_out,
+                              conv_bn_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(bn_saved_mean, bn_saved_mean, conv_bn_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(bn_saved_variance, bn_saved_variance,
+                              conv_bn_pattern);
+
+    // Find bias from BN
+    auto* bn_bias_var = scope->FindVar(bn_bias->Name());
+    auto* bn_bias_tensor = bn_bias_var->GetMutable<LoDTensor>();
+
+    // Get y (bias) variable
+    auto* bias_y_tensor =
+        scope->FindVar(bias_y->Name())->GetMutable<LoDTensor>();
+
+    PADDLE_ENFORCE_EQ(bias_y_tensor->dims(), bn_bias_tensor->dims());
+
```

Comment:
It's better to add some formulas here, like
https://github.com/PaddlePaddle/Paddle/blob/35b713c3fd63e98380efa0dd1848b9f9a8223af8/python/paddle/fluid/transpiler/inference_transpiler.py#L247-258

Reply:
I'm not sure where to add a block comment like this.